### PR TITLE
[CMakeLists] Remove unused SWIFT_INCLUDE_OUTPUT_INTDIR + comment out unnecessary logic

### PR DESCRIPTION
**Pull request by practicalswift**
`SWIFT_INCLUDE_OUTPUT_INTDIR` appears not to be unused (let me know if I am missing some indirect usage path :-)):

```
$ git grep SWIFT_INCLUDE_OUTPUT_INTDIR
$
```

### CODE REVIEW

**Comment by modocache** [comment]
Thanks so much for this. Trimming down the build scripts is really helpful. Obsolete variables used can lead people down rabbit holes--which happened to me just yesterday. :smiley:

**Comment by practicalswift** [comment]
@modocache Actually it was your PR #1410 that made me look into unused `CMakeLists` variables :-)

I wrote a small scripts that tries to find unused `CMakeLists` variables. I've found 18 candidates for deletion (including the one reported in #1410 and the two covered by this PR).

However, I'm a bit unsure about some of them so I'd love to have someone familiar with `cmake`/the general build process look in to each case to see if I'm missing some dependency/indirect usage. What is the best way to proceed? Should I submit a PR marked "WIP" and have it reviewed?

File: utils/CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 add_subdirectory(vim)
 
-if(SWIFT_INSTALL_TOOLS)
+# if(SWIFT_INSTALL_TOOLS)
```
Comment (gribozavr) [inline utils/CMakeLists.txt]: Just delete this code :)

**Comment by gribozavr** [comment]
This LGTM.

**Comment by practicalswift** [comment]
@gribozavr Updated and rebased as per your request :-)

File: utils/CMakeLists.txt (hunk 1 above)
Comment (modocache) [inline utils/CMakeLists.txt]: Agreed. If there is indeed something we can do better with regards to installing `gyb`, we should document that in a ticket on https://bugs.swift.org (along with *much* more detail than present here).

**Pull request merged**